tests/test_synchronization.py

```diff
@@ -524,3 +524,21 @@ async def acquire() -> None:
         await asyncio.sleep(0)
         task1.cancel()
         await asyncio.wait_for(task2, 1)
+
+    async def test_ordered_queue(self) -> None:
+        limiter = CapacityLimiter(1)
+        results = []
+        event = Event()
+
+        async def append(x: int, task_status: TaskStatus) -> None:
+            task_status.started()
```

Comment:
I think using `start()` instead of `start_soon()` is pointless here.

Reply:
`start_soon()` didn't seem to guarantee the order of the tasks for `trio`, though the other backends worked fine. Is that normal `trio` behaviour?